src/server/enrollment-system.js

```diff
@@ -932,21 +932,11 @@ function veteranToFinancialsInfo(veteran) {
   });
 
   const dependentFinancials = veteranToDependentFinancialsCollection(veteran);
-  let hasDependentFinancials = false;
-  if (dependentFinancials) {
-    hasDependentFinancials = _.compact(dependentFinancials.dependentFinancials.map((child) => { return child.incomes; })).length > 0;
-  }
   const spouseFinancials = veteranToSpouseFinancials(veteran);
-  const hasSpouseIncome = spouseFinancials && spouseFinancials.spouseFinancials.incomes;
-
-  const hasIncomeData = expenses || incomes || hasSpouseIncome || hasDependentFinancials;
-
-  if (!hasIncomeData) {
-    return undefined;
-  }
+  const discloseFinancials = veteran.discloseFinancialInformation;
 
   return {
-    incomeTest: booleanToIncomeTest(hasIncomeData),
+    incomeTest: booleanToIncomeTest(discloseFinancials),
```

Comment:
Ah, so you mean if I make the change you identified above, with returning undefined if `!veteran.discloseFinancialInformation`, then we assume it will be true if it reaches this point?